sequence_task_1/student/include/impl.h

```diff
@@ -8,33 +8,69 @@ class StaticPlaylist
 {
 public:
     /** @todo Member traits */
+    using value_type = Song_t;
+    using reference = Song_t&;
+    using const_reference = const Song_t&;
+    using iterator = typename Container::iterator;
+    using const_iterator = typename Container::const_iterator;
+    using difference_type = ptrdiff_t;
+    using size_type = size_t;
 
     /** @todo Iterators */
+    const_iterator begin() const {
+        return m_tracklist.cbegin();
+    }
+    const_iterator end() const {
+        return m_tracklist.cend();
+    }
 
     StaticPlaylist() = default;
 
     /** @todo Constructor from any reversible sequence container */
-
+    template<class T>
+    StaticPlaylist(const T& container_) {
+        m_tracklist.assign(container_.rbegin(), container_.rend());
+    }
     /** @todo Assignment from any reversible sequence container */
+    template<class T>
+    StaticPlaylist<Container, Song_t>& operator=(const T& container_){
+        m_tracklist.assign(container_.rbegin(), container_.rend());
 
+        return *this;
+    }
     /** @todo Add track from initializer */
     template<class... Args>
-    const Song_t& play(Args&&... songData);
+    const Song_t& play(Args&&... songData) {
+        m_tracklist.emplace_back(std::forward<Args>(songData)...);
+        return current();
+    }
 
     /** @todo Add track */
-    const Song_t& play(const Song_t& song);
+    const Song_t& play(const Song_t& song) {
+        m_tracklist.emplace_back(song);
```

Comment:
it's much closer to the "insert"/"push", rather than emplace - you're passing the lvalue